core/src/main/java/com/sequenceiq/distrox/v1/distrox/StackOperations.java

```diff
@@ -258,6 +258,14 @@ public FlowIdentifier putStart(NameOrCrn nameOrCrn, Long workspaceId) {
         return stackCommonService.putStartInWorkspace(nameOrCrn, workspaceId);
     }
 
+    public void rotateSaltPassword(NameOrCrn nameOrCrn, Long workspaceId) {
+        Stack stack = stackService.getByNameOrCrnAndWorkspaceIdWithLists(nameOrCrn, workspaceId);
+        if (flowLogService.isOtherFlowRunning(stack.getId())) {
+            throw new CloudbreakServiceException(String.format("Operation is running for stack '%s'. Please try again later.", stack.getName()));
+        }
```

Comment:
why is this necessary? if parallel run for the flow is not allowed, flow engine would reject it automatically

Reply:
The password rotation is not run in a flow, as it is a short operation

Reply:
That's not a good idea in my opinion. What happens if I send in multiple request with different values? I think this should be definitely part of a flow. Also salt calls with retries etc can take up a longer time than you expect and experience when testing the happy path.

Reply:
This operation does not have any input, just the stack crn, so I thought we do not need a flow, but I can add if you think it is needed.

Reply:
I think yes, as if I call it multiple times in a short time, the call can land on different pods, and you won't be able to tell, what is set on the instances and what is stored in vault in the end. Also these might not even be the same, so salt would be unusable and another rotation has to be invoked eventually. We had similar issue with bind user creation.